hal/halx86/legacy/bussupp.c

```diff
@@ -898,9 +902,9 @@ HalpDebugPciDumpBus(IN PBUS_HANDLER BusHandler,
             {
                 /* Copy the subvendor name into our buffer */
                 SubVendorName += strlen("\t\t0000 0000  ");
-                p = strpbrk(SubVendorName, "\r\n");
+                p = strpbrk(SubVendorName, NEWLINE);
                 Length = p - SubVendorName;
-                if (Length >= sizeof(bSubVendorName)) Length = sizeof(bSubVendorName) - 1;
+                Length = min(Length, sizeof(bSubVendorName) - 1);
```

Comment:
Couldn't we write:
Length = min(p - SubVendorName, sizeof(bSubVendorName) - 1);
at line 907 and entirely omit line 906.

Reply:
Yes we could, but since `min` (and `max`) is a macro: `#define min(a, b) ((a) < (b) ? (a) : (b))` for instance, it can evaluate more than once its arguments, so usually I avoid doing calculations within the min/max arguments (unless their result is known at compile-time, like the sizeof(xxx)-1).
(Also this makes the code line a bit shorter...)

Reply:
Your reasoning for trying to avoid the macros here does seem very legit to me. Thank you for explaining. Resolved.